lib/api/src/engine.rs

```diff
@@ -38,6 +38,12 @@ impl Engine {
         self.0.deterministic_id()
     }
 
+    /// Sets the hash algorithm
```

Comment:
This needs to live in `NativeEngineExt`, not here

Reply:
(remove it from here)

Reply:
You're right!